bin/propolis-server/src/lib/server.rs

```diff
@@ -975,23 +976,15 @@ async fn instance_issue_crucible_vcr_request(
     // to replace it.
     let vm_controller = rqctx.context().vm().await?;
     let mut spec = vm_controller.instance_spec().await;
+    let VersionedInstanceSpec::V0(v0_spec) = &mut *spec;
 
-    let (readonly, old_vcr) = {
-        let bes = &spec.backends.storage_backends.get(&disk_name);
-        if let Some(bes) = bes {
+    let (readonly, old_vcr_json) = {
+        let bes = &v0_spec.backends.storage_backends.get(&disk_name);
+        if let Some(StorageBackendV0::Crucible(bes)) = bes {
             let readonly = bes.readonly;
-            match &bes.kind {
-                StorageBackendKind::Crucible { req } => (readonly, req),
-                x => {
-                    let s = format!(
-                        "Invalid StorageBackendKind for VCR replacement: {:?}",
-                        x
-                    );
-                    return Err(HttpError::for_not_found(Some(s.clone()), s));
-                }
-            }
+            (readonly, &bes.request_json)
```

Comment:
Could we just return `(bes.readonly, &bes.request_json)` and avoid having a `readonly` local here?

Reply:
Oops. Fixed in 03e3d99.